test/distributed/org/apache/cassandra/distributed/action/GossipHelper.java

```diff
@@ -0,0 +1,462 @@
+    {
+        private final boolean joinRing;
+        private final Duration waitForBootstrap;
+        private final Duration waitForSchema;
+
+        public BootstrapAction()
+        {
+            this(true, Duration.ofMinutes(10), Duration.ofSeconds(10));
+        }
+
+        public BootstrapAction(boolean joinRing, Duration waitForBootstrap, Duration waitForSchema)
+        {
+            this.joinRing = joinRing;
+            this.waitForBootstrap = waitForBootstrap;
+            this.waitForSchema = waitForSchema;
+        }
+
+        public void accept(IInvokableInstance instance)
+        {
+            instance.appliesOnInstance((String partitionerString, String tokenString) -> {
+                IPartitioner partitioner = FBUtilities.newPartitioner(partitionerString);
+                List<Token> tokens = Collections.singletonList(partitioner.getTokenFactory().fromString(tokenString));
+                try
+                {
+                    SystemKeyspace.setBootstrapState(SystemKeyspace.BootstrapState.IN_PROGRESS);
+                    StorageService.instance.waitForSchema(Math.toIntExact(waitForSchema.toMillis()));
+                    PendingRangeCalculatorService.instance.blockUntilFinished();
+                    StorageService.instance.startBootstrap(tokens).get(waitForBootstrap.toMillis(), TimeUnit.MILLISECONDS);
+                    StorageService.instance.setUpDistributedSystemKeyspaces();
+                    if (joinRing)
+                        StorageService.instance.finishJoiningRing(true, tokens);
+                }
+                catch (Throwable t)
+                {
+                    throw new RuntimeException(t);
+                }
+
+                return null;
+            }).apply(instance.config().getString("partitioner"), instance.config().getString("initial_token"));
+        }
+    }
+
+    public static InstanceAction decomission()
+    {
+        return (target) -> {
+            target.runOnInstance(() -> {
+                try
+                {
+                    StorageService.instance.decommission(false);
+                }
+                catch (InterruptedException e)
+                {
+                    throw new RuntimeException();
+                }
+            });
+        };
+    }
+
+
+    public static VersionedApplicationState tokens(IInvokableInstance instance)
+    {
+        return versionedToken(instance, ApplicationState.TOKENS, (partitioner, tokens) -> new VersionedValue.VersionedValueFactory(partitioner).tokens(tokens));
+    }
+
+    public static VersionedApplicationState netVersion(IInvokableInstance instance)
+    {
+        return versionedToken(instance, ApplicationState.NET_VERSION, (partitioner, tokens) -> new VersionedValue.VersionedValueFactory(partitioner).networkVersion());
+    }
+
+    public static VersionedApplicationState unsafeAppState(IInvokableInstance instance, ApplicationState applicationState, Supplier<VersionedValue> supplier)
```

Comment:
this looks unused and the supplier isn't Serializable, so shouldn't callOnInstance fail?